src/libraries/System.Private.CoreLib/src/System/DateTime.Windows.cs

```diff
@@ -14,14 +14,39 @@ public static unsafe DateTime UtcNow
         {
             get
             {
+                long fileTime;
+                s_pfnGetSystemTimeAsFileTime(&fileTime);
+
                 if (s_systemSupportsLeapSeconds)
                 {
                     FullSystemTime time;
-                    GetSystemTimeWithLeapSecondsHandling(&time);
+
+                    if (Interop.Kernel32.FileTimeToSystemTime(&fileTime, &time.systemTime) != Interop.BOOL.FALSE)
```

Comment:
Some of the methods were split into multiple parts to make the FCalls work. I have undone the split since it is no longer needed.